scripts/check-peer-constraints.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+// +build ignore
+
+// check-peer-constraints checks that none of the core packages import peers directly.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+var race = flag.Bool("race", false, "load with race tag")
+var fail = flag.Bool("fail", true, "fail on violation")
+
+func main() {
+	flag.Parse()
+
+	var exitcode int
+
+	peers, err := load(
+		"storj.io/storj/satellite/...",
+		"storj.io/storj/storagenode/...",
+		"storj.io/storj/bootstrap/...",
+	)
+	if err != nil {
+		fmt.Printf("failed to load peers: %v\n", err)
+		os.Exit(1)
+	}
+
+	for _, source := range []string{
+		"storj.io/storj/pkg/...",
+		"storj.io/storj/lib/...",
+		"storj.io/storj/uplink/...",
+	} {
+		sources, err := load(source)
+		if err != nil {
+			fmt.Printf("failed to load %q: %v\n", source, err)
+			os.Exit(1)
+		}
+
+		if links(sources, peers) {
+			exitcode = 1
+		}
+	}
+
+	if *fail {
+		os.Exit(exitcode)
+	}
+}
+
+func load(names ...string) ([]*packages.Package, error) {
+	var buildFlags []string
+	if *race {
+		buildFlags = append(buildFlags, "-race")
+	}
+
+	return packages.Load(&packages.Config{
+		Mode:       packages.LoadImports,
+		Env:        os.Environ(),
+		BuildFlags: buildFlags,
+		Tests:      false,
```

Comment:
I think we might want to include tests as well.

Reply:
We a have a lot of system level tests. To test uplink fully we need to have a running storage node and satellite.

Reply:
Dang. Works for me. Additionally, if we ever have problems we can iterate on catching them here.